sql/core/src/test/scala/org/apache/spark/sql/streaming/resources/testCommitLogV1CrossVersion/testCommitLog

```diff
@@ -0,0 +1,2 @@
+v1
+{"nextBatchWatermarkMs":1,"stateUniqueIds":{}}
```

Comment:
Do we need to store it as a resource file? Can't we write a temp file from the code from a constant string?

Also, this file can be generated using class CommitMetadata so I don't know why we need to store it. It is a good idea to generate a file with `{"nextBatchWatermarkMs":1}` though, as what @brkyvz said.

Reply:
You could also write a constant string too. It just makes it easier to load the file from resources than to have to write a local file every single test.

Storing persisted data as a resource folder for backwards compatibility testing is a very common pattern though, that's why I'd prefer to follow that pattern

Reply:
>  It is a good idea to generate a file with {"nextBatchWatermarkMs":1} though,
This is covered in existing tests

Reply:
Oh great! Ignore my other comment then. I'm good with the current code.